bool status
---
int32 state
